src/backend/winit/mod.rs

```diff
@@ -308,18 +209,24 @@ where
         &mut self.renderer
     }
 
-    /// Bind the underlying window to the underlying renderer
+    /// Bind the underlying window to the underlying renderer.
+    ///
+    /// The `size` argument is used to resize the underlying renderer. When `None` no
+    /// resize is done.
     #[instrument(level = "trace", parent = &self.span, skip(self))]
     #[profiling::function]
     pub fn bind(&mut self) -> Result<(), crate::backend::SwapBuffersError> {
-        // apparently the nvidia-driver doesn't like `wl_egl_window_resize`, if the surface is not current.
-        // So the order here is important.
-        self.renderer.bind(self.egl.clone())?;
-
-        // Were we told to resize?
-        if let Some(size) = self.resize_notification.take() {
-            self.egl.resize(size.w, size.h, 0, 0);
+        // NOTE we must resize before making the current context current, otherwise the back
+        // buffer will be latched. Some nvidia drivers may not like it, but a lot of wayland
+        // software does the order that way due to mesa latching back buffer on each
+        // `make_current`.
```

Comment:
Urgh, so this order is explicitly required for correct behavior in mesa and the other for nvidia? That's stupid.

In that case we need a test for the driver here, because this order crashes the client on nvidia.

Reply:
Are you sure that it still crashes on nvidia that way? We use the order like that in alacritty for a while and it's not crashing. I think they changed it a while ago to not crash though.

Reply:
It might be fixed now, but I definitely remember debugging this a whole afternoon. What anvil does differently to most wayland clients is, that it will unbind surface/context again, since we might have multiple. So I am not sure, if that same bug is triggered by most other clients anyway, though I believe you with alacritty.

I will re-test it with this branch and an up-to-date driver, before we touch the code. I would love to be able to keep this order.

Reply:
@Drakulix were you able to verify whether nvidia works. We're doing basically what smithay does in alacritty without issues. Though, we don't unbind completely.

Reply:
I don't have a system running nvidia+x11 anymore, but under wayland at least this seems to be running the same on all my gpus. I call that good enough. If there actually was an issue here, they fixed it or it is x11 exclusive (which is probably broken anyway, because untested).